tsdb/engine/tsm1/encoding.go

```diff
@@ -0,0 +1,571 @@
+package tsm1
+
+import (
+	"encoding/binary"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/influxdb/influxdb/tsdb"
+)
+
+const (
+	// BlockFloat64 designates a block encodes float64 values
+	BlockFloat64 = 0
+
+	// BlockInt64 designates a block encodes int64 values
+	BlockInt64 = 1
+
+	// BlockBool designates a block encodes bool values
+	BlockBool = 2
+
+	// BlockString designates a block encodes string values
+	BlockString = 3
+)
+
+type Value interface {
+	Time() time.Time
+	UnixNano() int64
+	Value() interface{}
```

Comment:
We've had quite a few bugs around our overuse of `interface{}`. How about at least typing this like so:

`type ValueType interface{}`

and then pushing the decode functionality in `NewValue` onto that type?